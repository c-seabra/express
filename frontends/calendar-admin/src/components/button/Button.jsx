import React from 'react'

const Button = ({ className, onBtnClick, children }) => (
  <button className={className} onClick={e => onBtnClick ? onBtnClick() : e.preventDefault()}>{children}</button>
)
export default Button
