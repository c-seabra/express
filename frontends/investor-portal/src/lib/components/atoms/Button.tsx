/**
 * Button - atom
 *
 * Types
 *  Primary, Secondary
 *
 * States
 *  disabled, hovered
 */

import styled from 'styled-components'

export const Button = styled.button`
  display: inline-flex;
  font-size: 0.875rem;
  font-weight: 400;
  justify-content: center;
  align-items: center;
  min-width: 75px;
  padding: 0 32px;
  border-radius: 3px;
  border: 1px solid #0067e9;
  letter-spacing: 0;
  transition: all 0.3s ease-out;
  line-height: 24px;
  height: 36px;

  cursor: pointer;

  color: #ffffff;
  background-color: #0067e9;

  &:hover {
    border: 1px solid #2269bf;
    background-color: #2269bf;
  }

  &:focus {
    outline: none;
  }

  &:disabled {
    background-color: #cccccc;
    color: #888888;
    border-color: #c1c5c9;
    cursor: not-allowed;
    pointer-events: none;
  }
`

export const SecondaryButton = styled(Button)`
  color: #0067e9;
  background-color: #ffffff;

  &:hover {
    background-color: #2269bf;
    color: #ffffff;
  }
`

export const DestructiveButton = styled(Button)`
  color: #cb1977;
  background-color: #ffffff;
  border: 1px solid #cb1977;

  &:hover {
    background-color: #cb1977;
    color: #ffffff;
  }
`
export default Button
