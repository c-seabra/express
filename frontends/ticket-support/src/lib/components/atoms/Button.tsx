/**
 * Button - atom
 *
 * Types
 *  Primary, Secondary
 *
 * States
 *  disabled, hovered
 */

import React, { ReactElement } from 'react'
import styled from 'styled-components'

const StyledButton = styled.button`
  display: inline-flex;
  box-sizing: border-box;

  font-family: 'azo-sans-web', sans-serif;
  font-size: 14px;
  font-weight: normal;

  text-align: center;
  min-width: 75px;
  padding: 0 32px;
  border-radius: 3px;
  border: 1px solid #0067e9;
  letter-spacing: 0;
  transition: all 0.3s ease-out;
  color: #fff;
  background-color: #000;
  height: 50px;

  cursor: pointer;

  &:focus {
    outline: none;
  }

  &:disabled {
    background-color: #aaaaaa;
    color: #7d7d7d;
    border-color: #c1c5c9;
    cursor: not-allowed;
    pointer-events: none;
  }
`

export const PrimaryButton = styled(StyledButton)`
  color: #ffffff;
  background-color: #0067e9;

  &:hover {
    border: 1px solid: #2269bf;
    background-color: #2269bf;
  }
`

export const SecondaryButton = styled(StyledButton)`
  color: #0067e9;
  background-color: #ffffff;

  &:hover {
    background-color: #eee;
  }
`

type ButtonProps = {
  children?: ReactElement
  className?: string
  type?: string
}

type Props = ButtonProps | React.InputHTMLAttributes<HTMLButtonElement>

const Button = ({ children }: Props, props: any): ReactElement => {
  return <button {...props}>as{children}</button>
}

export default Button
