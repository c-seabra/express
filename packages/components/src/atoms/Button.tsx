/**
 * Button - atom
 *
 * Types
 *  Primary, Secondary
 *
 * States
 *  disabled, hovered
 */

import styled from 'styled-components';

export const Button = styled.button`
  display: inline-flex;
  font-size: 0.875rem;
  font-weight: 400;
  justify-content: center;
  align-items: center;
  min-width: 75px;
  padding: 0 32px;
  border-radius: 3px;
  border: 1px solid #0067e9;
  letter-spacing: 0;
  transition: all 0.3s ease-out;
  line-height: 24px;
  height: 36px;

  cursor: pointer;

  color: #ffffff;
  background-color: #0067e9;

  &:hover {
    border: 1px solid #2269bf;
    background-color: #2269bf;
  }

  &:focus {
    outline: none;
  }

  &:disabled {
    background-color: #cccccc;
    color: #888888;
    border-color: #c1c5c9;
    cursor: not-allowed;
    pointer-events: none;
  }
`;

export const SecondaryButton = styled(Button)`
  color: #0067e9;
  background-color: #ffffff;

  &:hover {
    background-color: #2269bf;
    color: #ffffff;
  }
`;

export const ErrorButton = styled(Button)`
  background-color: #e15554;
  border: 1px solid #e15554;

  &:hover {
    background-color: #de201f;
    border: 1px solid #de201f;
  }
`;

export const DestructiveButton = styled(Button)`
  color: #cb1977;
  background-color: #ffffff;
  border: 1px solid #cb1977;

  &:hover {
    background-color: #cb1977;
    color: #ffffff;
  }
`;

export const DisabledButton = styled(Button)`
  background-color: #949494;
  border: 1px solid #949494;

  &:hover {
    background-color: #6b6b6b;
    border: 1px solid #6b6b6b;
  }
`;

export const TextButton = styled(SecondaryButton)`
  border: none;
  background-color: white;
  padding: 0;

  &:hover {
    background-color: white;
    color: #2269bf;
    border: none;
  }
`;
