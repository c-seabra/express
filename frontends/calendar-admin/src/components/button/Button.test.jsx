import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('Button', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    React.useContext = jest.fn();
    useContextMock = React.useContext;
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  test('renders without props', () => {
    useContextMock.mockReturnValue({});
    const component = renderer.create(<Button />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders with props', () => {
    useContextMock.mockReturnValue({});
    const component = renderer.create(
      <Button className="button" onClick={() => null}>
        Some Text
      </Button>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
