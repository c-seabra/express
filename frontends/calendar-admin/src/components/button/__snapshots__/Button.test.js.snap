// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with props 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Some Text
</button>
`;

exports[`Button renders without props 1`] = `
<button
  className={undefined}
  onClick={[Function]}
/>
`;
